lib/elastic_record/as_document.rb

```diff
@@ -16,27 +16,52 @@ def as_partial_update_document
 
       changed_fields.each_with_object({}) do |field, result|
         if field_mapping = mappings[field]
-          result[field] = elastic_search_value field, field_mapping
+          result[field] = value_for_elastic_search field, field_mapping
         end
       end
     end
 
-    def elastic_search_value(field, mapping)
+    def value_for_elastic_search(field, mapping)
       value = try field
       return if value.nil?
 
-      value = case mapping[:type]&.to_sym
-              when :object
-                value.respond_to?(:as_search_document) ? value.as_search_document : value
-              when :nested
-                value.map(&:as_search_document)
-              else
-                value
-              end
+      value =
+        case mapping[:type]&.to_sym
+        when :object
+          value_for_elastic_search_object(value)
+        when :nested
+          value.map { |entry| value_for_elastic_search_object(entry) }
+        when :integer_range, :float_range, :long_range, :double_range, :date_range
+          value_for_elastic_search_range(value)
+        else
+          value
+        end
 
       if value.present? || value == false
         value
       end
     end
+
+    def value_for_elastic_search_object(object)
+      object.respond_to?(:as_search_document) ? object.as_search_document : object
+    end
+
+    def value_for_elastic_search_range(range)
+      ordered = ordered_range(range)
+
+      gte = ordered.begin unless ordered.begin == -Float::INFINITY
+      lte = ordered.end unless ordered.end == Float::INFINITY
+      {'gte' => gte, 'lte' => lte}
+    end
+
+    private
+
+      def ordered_range(range)
+        if range.begin > range.end
+          range.end..range.begin
+        else
+          range
+        end
+      end
```

Comment:
This is silly. Why support this edge case? Is there a use case for it?

Reply:
And if required, you could use min/max with fewer lines of code in the method above :)

Reply:
All elasticsearch ranges are ruby ranges.  Some ruby ranges are elasticsearch ranges.

Should it be possible to initialize a document with values that won't serialize?

> you could use min/max with fewer lines of code in the method above :)

Surprisingly you can't when `begin > end`!  [Range#min](https://ruby-doc.org/core-2.2.0/Range.html#method-i-min)

Reply:
What test case shows this problem?

Reply:
> What test case shows this problem?

https://github.com/data-axle/elastic_record/pull/71/files#diff-32fef10a3489f3c4d2fd0ccf18043c8cR66

Reply:
I see. Change superstore to make the range nil if begin > end. Allow ES to crash here if the data is bad.